lib/manageiq/messaging/kafka/client.rb

```diff
@@ -85,6 +85,11 @@ def initialize(options)
           connection_opts = {:"bootstrap.servers" => hosts.join(',')}
           connection_opts[:"client.id"] = options[:client_ref] if options[:client_ref]
 
+          options_copy = Hash[options]
+          [:port, :host, :hosts, :encoding, :protocol, :client_ref].each { |k| options_copy.delete(k) }
```

Comment:
```suggestion
          options_copy = options.except(:port, :host, :hosts, :encoding, :protocol, :client_ref)
```

Reply:
@bdunne Thank you. I think I can avoid using the extra `options_copy` variable all together.